packages/shared/src/components/sidebar/common.tsx

```diff
@@ -135,18 +137,22 @@ export const ItemInner = ({
   active,
 }: ItemInnerProps): ReactElement => {
   const Icon = shouldShowLabel ? ItemInnerIcon : ItemInnerIconTooltip;
+  const { sidebarExpanded } = useContext(SettingsContext);
+
   return (
     <>
       <Icon {...item} active={active} />
-      <span
-        className={classNames(
-          'flex flex-1 text-left transition-opacity truncate flex-row items-center',
-          shouldShowLabel ? 'opacity-100 delay-150' : 'opacity-0',
-          item?.className?.text,
-        )}
-      >
-        {item.title}
-      </span>
+      {sidebarExpanded && (
```

Comment:
I think this will take away the current animation on our Sidebar. Though not sure if it is high priority to be kept. cc @rebelchris 

The `shouldShowLabel` should handle whether this label is actually showed, so we might want to pass this `sidebarExpanded` property in the items though I believe it is already included. I could be wrong though.

Reply:
Oh well spotted, yes please make sure this doesn't remove the animation.

Reply:
We used to have a hook to manage these scenarios and allow animation on dynamically loaded content. Kindly check the link below if it would be of any use to you:
packages/shared/src/hooks/useDynamicLoadAnimated.ts

Reply:
Looked into this.. honestly, with `150ms` delay for that opacity animation it is almost invisible to the human eye and also not worth it. Even with this hook I don't think including that much JavaScript is worth it. 

The main reason I did this change is so I can continue to use `rightIcon` prop for the changelog popup anchor. If we wanna retain this animation I think I'll have to add the dot somewhere else because obviously `rightIcon` is not a good choice since it is not visible when sidebar is hidden. 

<img width="250" alt="image" src="https://user-images.githubusercontent.com/9803078/223456250-e4c0e208-fe79-4924-b1f5-5c6c73319564.png">

Reply:
I agree we must put the dot somewhere else. I think we don't even have to use the `rightIcon` anymore since our reference point is the list item. I highly suggest making the `BaseTooltipContainer` accept an `arrow` as `ReactNode` it would be easier on every part as the dot will always be attached to the arrow, and we won't have to worry about whether it is going to be displayed or not since it will be part of the popup that is always on top of the screen.

Reply:
I think animations are those small things that are not talked about but are well appreciated by the users. For me, I'd love to keep it, but feel free to talk about it with @rebelchris.

Reply:
Agree in general, of course wanna try and do minimal impact on existing features, it is tricky when this kind of elements are introduced.

Reply:
I think it's quite noticeable to have and we should keep that behaviour.
Note on opening you see if quite well.

https://user-images.githubusercontent.com/554874/223675861-5335ef4e-612e-484a-b185-56ab8d9a4573.mp4

Reply:
@sshanzel @rebelchris This should be good now, made changes to BaseTooltip and refactored other usages in c4076a411f3f7fce13b33df28129bf1186a0747c:
* added ArrowComponent for custom arrows
* remove arrowClassname, we don't need it because we can just add classes to custom ArrowComponent
* renamed `arrow` to `showArrow` - I think it better represents what prop means
  * it could have been done with passing `null` to `ArrowComponent` and having only one prop but I think it is more clear this way